src/pages/networks/forms/NetworkFormMenu.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, useEffect, useState } from "react";
+import MenuItem from "pages/instances/forms/FormMenuItem";
+import { Button, useNotify } from "@canonical/react-components";
+import { updateMaxHeight } from "util/updateMaxHeight";
+import useEventListener from "@use-it/event-listener";
+import { FormikProps } from "formik/dist/types";
+import { NetworkFormValues } from "pages/networks/forms/NetworkForm";
+
+export const MAIN_CONFIGURATION = "Main configuration";
+export const BRIDGE = "Bridge";
+export const DNS = "DNS";
+export const IPV4 = "IPv4";
+export const IPV6 = "IPv6";
+export const YAML_CONFIGURATION = "YAML configuration";
+
+interface Props {
+  active: string;
+  setActive: (val: string) => void;
+  formik: FormikProps<NetworkFormValues>;
+}
+
+const NetworkFormMenu: FC<Props> = ({ active, setActive, formik }) => {
+  const notify = useNotify();
+  const [isAdvancedOpen, setAdvancedOpen] = useState(false);
+  const menuItemProps = {
+    active,
+    setActive,
+  };
+
+  const hasName = formik.values.name.length > 0;
+  const hasOvnTypeMissingUplink =
+    formik.values.type === "ovn" && (formik.values.network?.length ?? 0) < 1;
+  const disableReason = hasName
+    ? hasOvnTypeMissingUplink
+      ? "Please select an uplink network"
+      : undefined
+    : "Please enter a network name to enable this section";
```

Comment:
I like this feature, it makes a lot of sense. However, my proposal, to make it more reliable, would be to bind this not only to "any" name being entered in the name field, but to a valid name, so that we don't have an inconsistent state where we can input advanced config but cannot create the network because the current name is not valid. Also, shouldn't we apply the same logic to profile creation as well?

Reply:
I add it on the profile creation as well.
I'd avoid validating the name to keep it simple. If the name is invalid, the input for it should show the validation error.